share/spack/setup-env.csh

```diff
@@ -41,7 +41,9 @@ if ($?SPACK_ROOT) then
 
     # Set up modules and dotkit search paths in the user environment
     # TODO: fix SYS_TYPE to something non-LLNL-specific
-    _spack_pathadd DK_NODE    "$_spack_share_dir/dotkit/$SYS_TYPE"
-    _spack_pathadd MODULEPATH "$_spack_share_dir/modules/$SYS_TYPE"
+    if ($?SYS_TYPE) then
+        _spack_pathadd DK_NODE    "$_spack_share_dir/dotkit/$SYS_TYPE"
+        _spack_pathadd MODULEPATH "$_spack_share_dir/modules/$SYS_TYPE"
+    endif
```

Comment:
Sometimes good old fashioned copy-n-paste is easier than messing with cherry picking. I would just undo these changes, push a commit, switch back to develop, create a new branch, and redo the change.

The problem with including unrelated commits in a PR is you never know when someone will disagree with the change. If they're in 2 PRs, neither change can hold up the other PR.